rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,257 @@
+
+        """
+        best_match = ''
+        best_match_section = ''
+        for sec_name, sec_content in six.iteritems(rebase_spec_file.spec_content.sections):
+            if sec_name.startswith('%files'):
+                for line in sec_content:
+                    new_best_match = difflib.get_close_matches(file, [best_match, MacroHelper.expand(line)])
+                    if new_best_match:
+                        # the new match is a closer match
+                        if new_best_match[0] != best_match:
+                            best_match = new_best_match[0]
+                            best_match_section = sec_name
+
+        return best_match_section or rebase_spec_file.get_main_files_section()
+
+    @classmethod
+    def _correct_missing_files(cls, rebase_spec_file, files):
+        """Adds files found in sources to the SPEC file."""
+        macros = [m for m in rebase_spec_file.macros if m['name'] in MacroHelper.MACROS_WHITELIST]
+        macros = MacroHelper.expand_macros(macros)
+        # ensure maximal greediness
+        macros.sort(key=lambda k: len(k['value']), reverse=True)
+
+        result = collections.defaultdict(lambda: collections.defaultdict(list))
+
+        for file in files:
+            section = cls._get_best_matching_files_section(rebase_spec_file, file)
+            substituted_path = MacroHelper.substitute_path_with_macros(file, macros)
+            rebase_spec_file.spec_content.sections[section].insert(0, substituted_path)
+            result['added'][section].append(substituted_path)
+            logger.info('Added %s to %s section', substituted_path, section)
+
+        return result
+
+    @classmethod
+    def _correct_deleted_files(cls, rebase_spec_file, files):
+        """Removes files missing in sources from the SPEC file."""
+        result = collections.defaultdict(lambda: collections.defaultdict(list))
+        for sec_name, sec_content in six.iteritems(rebase_spec_file.spec_content.sections):
+            if sec_name.startswith('%files'):
+                i = 0
+                while i < len(sec_content):
+                    original_line = sec_content[i].split()
+                    if not original_line:
+                        i += 1
+                        continue
+                    split_line = original_line.copy()
+                    directives = []
+                    prepend_macro = None
+                    for element in reversed(split_line):
+                        if element in cls.FILES_DIRECTIVES:
+                            if cls.FILES_DIRECTIVES[element]:
+                                prepend_macro = cls.FILES_DIRECTIVES[element]
+                            directives.insert(0, element)
+                            split_line.remove(element)
+
+                    if prepend_macro:
+                        split_line = [os.path.join(prepend_macro, file) for file in split_line]
+                    split_line = [MacroHelper.expand(f) for f in split_line]
```

Comment:
Why `f` here and `file` above? Single-char variables in list comprehensions are preferred. Also, why not `p` instead of `f`? It's not always a file, but it's always a path.